### PR TITLE
feat: add artifact upload for coverage e2e

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**

**What this PR does / why we need it**:
Ensures no regression for #2815 
**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .github/workflows/check-coverage.yml [hunk 1]
```diff
@@ -67,6 +67,11 @@ jobs:
           echo "Running e2e tests"
           cd e2e
           go test -e2eMode="coverage" -timeout 60m -v -run Test_All ./
+      - name: Upload e2e artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: scenario-logs
+          path: e2e/scenario-logs
```
Comment (cameronmeissner) [inline .github/workflows/check-coverage.yml:74]: something I was meaning to do on the ADO side - can we upload a new bundle that doesn't include scenario-logs/*/vmssId.txt? those text files are only there such that we can ensure the ephemeral VMSS are deleted (we have an 'ensure vmss deletion step in the ADO pipeline that reads the ids from those text files and does 'az delete --ids ...' on them

I'm also realizing now that we don't have the same sort of 'ensure vmss deletion' step on the GH action side...

we should really choose one and pivot towards that

Comment (cameronmeissner) [reply .github/workflows/check-coverage.yml:74]: might also be worth leaving those text files lying around so we can use them to manually delete stuff via az cli if needed....

I'll leave it up to you

**Review by cameronmeissner — APPROVED** [review]

File: .github/workflows/check-coverage.yml (hunk 1 above)
Comment (cameronmeissner) [reply .github/workflows/check-coverage.yml:74]: (can also just go use the portal to delete... :D)

Comment (UtheMan) [reply .github/workflows/check-coverage.yml:74]: I will be looking into how i could combine ADO running e2e and gh action taking results of that and handling everything else

**Review by alexeldeib — APPROVED** [review]

**Pull request merged**